subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CompileJavaBuildOperationReportingCompiler.java

```diff
@@ -88,6 +88,18 @@ private DefaultAnnotationProcessorDetails toAnnotationProcessorDetails(Annotatio
     }
 
     private static class Details implements CompileJavaBuildOperationType.Details {
+
+        private final Object taskBuildOperationId;
```

Comment:
Build operations already form a hierarchy, so an observer can already tell that this is the child of a certain Java compile task. So this info is redundant.

Reply:
Done in 9a64bb5cb59.